File: 简单/Java/leetCode/src/main/java/com/kurumi/leetCode/动态规划/LeetCode_0746.java
// 746. 使用最小花费爬楼梯
package com.kurumi.leetCode.动态规划;

/**
 * @author: kurumi
 * @date: 2024-10-16 21:58:27
 */
public class LeetCode_0746 {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp = new int[cost.length + 1];
        dp[0] = 0;
        dp[1] = 0;
        for (int i = 2; i <= cost.length; i++) {
            dp[i] = Integer.min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
        }
        return dp[cost.length];
    }
}

File: 卡码/代码随想录/C++/动态规划/dp/01-基础题目/LeetCode_0746.cpp
//746. 使用最小花费爬楼梯
#include <iostream>
#include <algorithm>
using namespace std;

class Solution {
public:
	int minCostClimbingStairs(vector<int>& cost) {
		int dp[cost.size()+1];
		dp[0]=0,dp[1]=0;
		for(int i=2;i<=cost.size();i++){
			dp[i]=min(dp[i-1]+cost[i-1],dp[i-2]+cost[i-2]);
		}
		return dp[cost.size()];
	}
};

int main(){
  system("pause");
  return 0;
}
